File: python/merge_sort.py
def merge(left_side, right_side):
    array = []
    left = 0
    right = 0
    while left < len(left_side) and right < len(right_side):
        left_el = left_side[left]
        right_el = right_side[right]
        if left_el < right_el:
            array.append(left_el)
            left += 1
        else:
            array.append(right_el)
            right += 1
    array += left_side[left:]
    array += right_side[right:]
    return array


def sort(array):
    if (len(array) == 1):
        return array
    mid = len(array) // 2
    left_side = sort(array[:mid])
    right_side = sort(array[mid:])
    return merge(left_side, right_side)

File: ruby/merge_sort.rb

def merge(left_side, right_side)
  arr = []
  left = right = 0
  max_left = left_side.size - 1
  max_right = right_side.size - 1

  while left <= max_left and right <= max_right
    if left_side[left] <= right_side[right]
      arr << left_side[left]
      left += 1
    else
      arr << right_side[right]
      right += 1
    end
  end
  arr += Array(left_side[left..-1])
  arr += Array(right_side[right..-1])
end

def merge_sort(arr)
  return arr if arr.size < 2
  mid = arr.size / 2
  left_side = merge_sort(arr[0...mid])
  right_side = merge_sort(arr[mid..-1])
  merge(left_side, right_side)
end
